client/my-sites/domains/domain-management/settings/cards/domain-redirect-card.tsx

```diff
@@ -61,41 +110,33 @@ const DomainRedirectCard = ( props: DomainRedirectCardProps & PropsFromRedux ) =
 			setIsValidUrl( false );
 			return;
 		}
+
 		setIsValidUrl( true );
 	};
 
 	const handleDelete = () => {
-		if ( ! redirect?.domainRedirectId || ! redirect?.targetHost ) {
+		if ( isLoading || ! redirect ) {
 			return;
 		}
+		deleteDomainRedirect();
+	};
 
-		setTargetUrl( '' );
```

Comment:
Moved this up to the onSuccess handler so we don't temporarily show a clickable save button while refreshing the form.